src/zjs_ocf_client.c

```diff
@@ -117,175 +117,90 @@ static char *create_url(const char *uuid, const char *path)
-            size = oc_double_array_size(rep->value.array);
-            if (size > 0) {
-                ZVAL array = jerry_create_array(size);
-                for (i = 0; i < size; i++) {
-                    ZVAL val = jerry_create_boolean(b_array[i]);
-                    jerry_set_property_by_index(array, i, val);
-                    DBG_PRINT("%d \n", b_array[i]);
-                }
-                zjs_obj_add_object(prop_object, oc_string(rep->name), array);
-            }
-            DBG_PRINT("]\n");
-        case STRING_ARRAY:
-            DBG_PRINT("[ ");
-            u32_t size = oc_string_array_get_allocated_size(rep->value.array);
-            if (size > 0) {
-                ZVAL array = jerry_create_array(size);
-                for (i = 0; i < size; i++) {
-                    if (oc_string_array_get_item_size(rep->value.array, i) > 0) {
-                        ZVAL val = jerry_create_string(oc_string_array_get_item(rep->value.array, i));
-                        jerry_set_property_by_index(array, i, val);
-                        DBG_PRINT("%s \n", oc_string_array_get_item(rep->value.array, i));
-                    }
-                }
-                zjs_obj_add_object(prop_object, oc_string(rep->name), array);
-            }
-            DBG_PRINT("]\n");
-            break;
-        case OBJECT:
-            DBG_PRINT("{ Object }\n");
-            break;
-        default:
-            break;
-        }
-        rep = rep->next;
+    for (i = 0; i < p; i++) {
+        ZJS_PRINT("\t");
     }
-
-    return prop_object;
 }
 
-#ifdef DEBUG_BUILD
-static void print_props_data(oc_client_response_t *data)
+static void print_props_data(int prefix, oc_rep_t *data)
 {
+    int p = prefix;
     int i;
-    oc_rep_t *rep = data->payload;
+    oc_rep_t *rep = data;
     while (rep != NULL) {
-        ZJS_PRINT("Type: %u, Key: %s, Value: ", rep->type,
-                  oc_string(rep->name));
         switch (rep->type) {
         case BOOL:
-            ZJS_PRINT("%d\n", rep->value.boolean);
+            do_print("%s: %d\n", oc_string(rep->name), rep->value.boolean);
             break;
         case INT:
-            ZJS_PRINT("%d\n", rep->value.integer);
-            break;
-        case DOUBLE:
-            ZJS_PRINT("%lf\n", rep->value.double_p);
+            do_print("%s: %d\n", oc_string(rep->name), rep->value.integer);
             break;
         case BYTE_STRING:
         case STRING:
-            ZJS_PRINT("%s\n", oc_string(rep->value.string));
+            do_print("%s: %s\n", oc_string(rep->name), oc_string(rep->value.string));
+            break;
+        case OBJECT:
+            do_print("%s: {\n", oc_string(rep->name));
+            print_props_data(p + 1, rep->value.object);
+            do_print("}\n");
+            break;
+        case STRING_ARRAY:
+            do_print("%s: [", oc_string(rep->name));
+            for (i = 0;
+                    i < oc_string_array_get_allocated_size(rep->value.array);
```

Comment:
align here

Reply:
I think it is right? its indented 4 spaces past the `i = 0` on the next line. That `oc_string_array_get_allocated_size()` is just quite a long function name.

Reply:
ok, i would usually indent 4 spaces from the for, or align the i with i = 0, but it's fine

Reply:
Huh? No, this looks wonky. 
How about call get_allocated_size on the previous line w/ a tmp var and check against that var to fit on one line.

That will also save you calling that function over and over again, I assume it always gives the same answer?

Do that in the cases below, too.